src/components/ui/dashboard/select-coaching-session.tsx

```diff
@@ -58,19 +51,18 @@ export function SelectCoachingSession({
   const [organizations, setOrganizations] = useState<Organization[]>(
     defaultOrganizations()
   );
-  useEffect(() => {
-    async function loadOrganizations() {
-      if (!userUUID) return;
+  const { userUUID } = useAuthStore((state) => state);
```

Comment:
I still don't prefer to break encapsulation of the SelectionCoachingSession component by directly using `useAuthStore()` instead of passing it as a property one level deep. [Here's a good explanation of my thought process from Perplexity.ai](https://www.perplexity.ai/search/is-it-proper-J6l4Nqz3TECfYyL6DC9I8A) summarizing other sites on this.

Here's my own thought process: I think it's a logical practice to retrieve state at the page level and pass user session state like userUUID down to components because components _are_ reusable and pages are not. But a component like this one becomes less reusable when tightly coupling the AuthStore into the innards of the component.